### ISSUE TITLE
Error when towny stops on a restart

### ISSUE THREAD

**Issue by Mundo-Minecraft** [TownyAdvanced/Towny#7385]
### What steps will reproduce the problem?

In the previous report (https://github.com/TownyAdvanced/Towny/issues/7372) I got a java error when restarting the server, now the error no longer appears. But the result is the same, towny does not close correctly and leaves the server "blocked" 

### What is the expected output?

A successful reboot

### Towny version

0.100.2.6

### Server version

Paper version git-Paper-496 (MC: 1.20.4)

### Please use Pastebin.com to link the following files

1. Your full server startup from the logs\latest.log: https://pastebin.com/0666NKYS
2. Your Towny config.yml: https://pastebin.com/XDTASugH
3. Your townyperms.yml: https://pastebin.com/H5uTJcsS
4. Your log's error: https://pastebin.com/PrRJbJYT

(optional) If this is to do with permissions,
5. Your permissions file:

(optional) If this is to do with chat,
6. Your chatconfig.yml:
7. Your channels.yml:

**Comment by Warriorrrr** [comment]
Please provide a thread dump from while the server is blocked

**Comment by Mundo-Minecraft** [comment]
> Please provide a thread dump from while the server is blocked

Hello, I am looking for the best way to do that but I find unreliable information. How can I provide that information?

Thank you.

**Comment by Warriorrrr** [comment]
Use the `jps` command to get process ids for currently running java processes, then use `jstack <process id>` for the server process

**Comment by Mundo-Minecraft** [comment]
Hello, when I write jps it tells me 7920 but when I write jstack 7920 it says "the target process 7920 does not respond within 10500 ms or HotSpot VM not loaded" so everything leaves this log

https://pastebin.com/SGp0TBxP

If I use the pid that appears when doing top, leave this other log

https://pastebin.com/TAtfB9pL

All of this has been with java 22, since I just updated the server to java 22 since minecraft 1.20.5 requires 21 (I had java 17).

Thx.

**Comment by Mundo-Minecraft** [comment]
It's already solved, I'll comment on what happened in case it happens to someone else. I have had Towny installed for 2 years (I restarted worlds 2 years ago) although I have been using it since 2013. 
I transferred the mysql to flatfile and saw that everything was working correctly (I didn't imagine it could be the database). Then I went from flatfile to mysql and now it works correctly and the server restarts fine...


It is very strange, I only went from mysql to flatfile and from flatfile to mysql... but it is solved

Greetings.
